include/shell.h

```diff
@@ -60,6 +60,10 @@ class BatchFile {
 	std::shared_ptr<BatchFile> prev = {}; // shared with Shell.bf
 	std::unique_ptr<CommandLine> cmd = {};
 	std::string filename{};
+
+private:
+	[[nodiscard]] std::string ExpandedBatchLine(std::string_view line);
```

Comment:
Is the `line` sv being re-assigned inside the function?

If not, suggest making it read-only at the API level w/ `const` because this helps readers understand the intent.

Reply:
+1

Reply:
I'm OK with that, and I see both points.

As a caller of the function prototype, my IDE expands the prototype and I see it takes in a `std::string_view`: Ok, I can rest assured my string won't be modified - that's as much as I need to know.

As a co-maintainer of the function body, `const` on the  arguments (yes, even scalars) tells me how the author intends to use the variable, and how I need to track it in my short-term memory.  

Non-`const` (even on scalars) in the body args says to the reader _"the author will/may/can re-assign this arg any point in the body, like a temporary variable in the code, so be on the look out for that."_

`const` (even on scalars) in the body args says to the reader _"the author is saying (and proving) that the arg's value stays the same throughout the body, so dear reader, you don't need to keep track of this one."

So if you really don't want to const the prototypes: OK, I get it, and as a caller I don't need to know more.

But on the function bodies: yes, please const (even sclars), because it's we're talking about communicating intention to readers/reviewers/maintainers - and `const` does the job very well.